components/tikv_util/src/resizable_threadpool.rs

```diff
@@ -1,161 +1,359 @@
-use tokio::{io::Result as TokioResult, runtime::Runtime};
-
-/// DaemonRuntime is a "background" runtime, which contains "daemon" tasks:
-/// any task spawn into it would run until finish even the runtime isn't
-/// referenced.
-pub struct DaemonRuntime(Option<Runtime>);
-
-impl DaemonRuntime {
-    /// spawn a daemon task to the runtime.
-    pub fn spawn(self: &Arc<Self>, f: impl Future<Output = ()> + Send + 'static) {
-        let wkr = self.clone();
-        self.0.as_ref().unwrap().spawn(async move {
-            f.await;
-            drop(wkr)
-        });
-    }
+use tokio::{
+    io::Result as TokioResult,
+    runtime::{Builder, Runtime},
+    time::interval,
+};
 
-    /// create a daemon runtime from some runtime.
-    pub fn from_runtime(rt: Runtime) -> Arc<Self> {
-        Arc::new(Self(Some(rt)))
-    }
+struct DeamonRuntime {
+    inner: Option<Runtime>,
+    task_count: Arc<AtomicUsize>,
 }
 
-impl Drop for DaemonRuntime {
+impl Drop for DeamonRuntime {
     fn drop(&mut self) {
-        if let Some(runtime) = self.0.take() {
+        if let Some(runtime) = self.inner.take() {
             runtime.shutdown_background();
         }
     }
 }
 
+#[derive(Clone)]
+pub struct DeamonRuntimeHandle {
+    inner: Weak<DeamonRuntime>,
+}
+
+impl DeamonRuntimeHandle {
+    pub fn spawn<Fut>(&self, fut: Fut)
+    where
+        Fut: Future<Output = ()> + Send + 'static,
+        {
+            let rc_runtime = match self.inner.upgrade() {
+                Some(runtime) => runtime,
+                None => return,
+            };
+        
+            let inner = match rc_runtime.inner.as_ref() {
+                Some(inner) => inner,
+                None => return,
+            };
+        
+            let handle = inner.handle().clone();
+            let task_count = rc_runtime.task_count.clone();
+        
+            task_count.fetch_add(1, Ordering::SeqCst);
+            
+            handle.spawn(async move {
+                fut.await;
+                task_count.fetch_sub(1, Ordering::SeqCst);
+            });
+        }
+
+    pub fn block_on<Fut>(&self, fut: Fut) -> Option<Fut::Output>
+    where
+        Fut: Future,
+    {
+        let (handle, task_count) = {
+            let rc_runtime = self.inner.upgrade()?;
+            let inner = rc_runtime.inner.as_ref()?;
+            let handle = inner.handle().clone();
+            let task_count = rc_runtime.task_count.clone();
+            (handle, task_count)
+        };    
+    
+        task_count.fetch_add(1, Ordering::SeqCst);
+    
+        Some(handle.block_on(async move {
+            let output = fut.await;
+            task_count.fetch_sub(1, Ordering::SeqCst);
+            output
+        }))
+    }
+}
+
 pub struct ResizableRuntime {
     pub size: usize,
+    count: usize,
     thread_name: String,
-    pool: Option<Arc<DaemonRuntime>>,
+    current_runtime: Arc<DeamonRuntime>,
+    used_runtime: Arc<Mutex<Vec<Arc<DeamonRuntime>>>>,
+    keeper: Option<Runtime>,
```

Comment:
just make `keeper` a `DaemonRuntime`, and remove `impl Drop for ResizableRuntime`.